From mboxrd@z Thu Jan  1 00:00:00 1970
Return-Path: <linux-kernel-owner@vger.kernel.org>
X-Spam-Checker-Version: SpamAssassin 3.4.0 (2014-02-07) on
	aws-us-west-2-korg-lkml-1.web.codeaurora.org
Received: from vger.kernel.org (vger.kernel.org [23.128.96.18])
	by smtp.lore.kernel.org (Postfix) with ESMTP id 9D836C25B50
	for <linux-kernel@archiver.kernel.org>; Tue, 24 Jan 2023 11:05:59 +0000 (UTC)
Received: (majordomo@vger.kernel.org) by vger.kernel.org via listexpand
        id S233787AbjAXLF6 (ORCPT <rfc822;linux-kernel@archiver.kernel.org>);
        Tue, 24 Jan 2023 06:05:58 -0500
Received: from lindbergh.monkeyblade.net ([23.128.96.19]:37368 "EHLO
        lindbergh.monkeyblade.net" rhost-flags-OK-OK-OK-OK) by vger.kernel.org
        with ESMTP id S233831AbjAXLFl (ORCPT
        <rfc822;linux-kernel@vger.kernel.org>);
        Tue, 24 Jan 2023 06:05:41 -0500
Received: from out4-smtp.messagingengine.com (out4-smtp.messagingengine.com [66.111.4.28])
        by lindbergh.monkeyblade.net (Postfix) with ESMTPS id F40FD3C21;
        Tue, 24 Jan 2023 03:05:31 -0800 (PST)
Received: from compute6.internal (compute6.nyi.internal [10.202.2.47])
        by mailout.nyi.internal (Postfix) with ESMTP id 268D05C092A;
        Tue, 24 Jan 2023 06:05:31 -0500 (EST)
Received: from imap50 ([10.202.2.100])
  by compute6.internal (MEProxy); Tue, 24 Jan 2023 06:05:31 -0500
DKIM-Signature: v=1; a=rsa-sha256; c=relaxed/relaxed; d=readahead.eu; h=
        cc:cc:content-type:date:date:from:from:in-reply-to:message-id
        :mime-version:reply-to:sender:subject:subject:to:to; s=fm3; t=
        1674558331; x=1674644731; bh=tEYZlf43dWAmtICS4wUuBE8sXSOA8yFH1Yj
        m+gpaNlY=; b=txsl7iatIcaamMHLXf8zCa7fxGJWLBs3beJ3qWIPLnrmhAR8mkd
        TGncrksVfnrw2IH2M/tyfIJCgC+nJ2Pm1iy1A7gaWiuANkGwaRJGigsX3QCBJzrH
        yncwMBb1uwnu3RelAwAk7o1OiRQwSKH/b6AsSG4yhPesLEHrRnjXhqMF7gNY86op
        N6yc5JlcERAXkKveXq53ET0NlokxxY1gT9JNjUPO0lIFHv1Cg7SlYLIWuV/CB7t8
        +ZM45RFqA5vKnoSaS54REFYYok94sAUyknyn/YMmVpxde5F7u4Fipc1Zi3ycIffi
        8L9FNnObaQIRZQkhqdnvDxmtPeN25ret+fQ==
DKIM-Signature: v=1; a=rsa-sha256; c=relaxed/relaxed; d=
        messagingengine.com; h=cc:cc:content-type:date:date:feedback-id
        :feedback-id:from:from:in-reply-to:message-id:mime-version
        :reply-to:sender:subject:subject:to:to:x-me-proxy:x-me-proxy
        :x-me-sender:x-me-sender:x-sasl-enc; s=fm3; t=1674558331; x=
        1674644731; bh=tEYZlf43dWAmtICS4wUuBE8sXSOA8yFH1Yjm+gpaNlY=; b=I
        IRMCaC8HW9R8RhfwAjJzkzQR9L9J+v6Z5833TBT9/nk+2IOv0HkaL3Q+ql4QpciA
        casH/tqUAhLB6b30QuZQPKIt6BTZdrMibpe/ORcLFWVDzVhXkqIRpt7uWNI5Nb+x
        cDhl8Bt9qxMIr379A0pHnPisMvXawP4qouNJjTgfrn4FCVzeaXUsal+IOx8TQmUu
        OZDtoHQvYfINJ0evvvztAviS6UCEE+hcOeNl9cd0dUpCX8gXpEjl66Uvssgc/310
        sX8GzGGbP2ytFh10P4KFzkvbcazKNFt7iF0EzJpywSGFYqSS8wfN8efiVqFwS+HK
        DeJ/rp6N4q22vmjwbObqg==
X-ME-Sender: <xms:ervPY9PqKQxmW8F_Hw3upBubqdBeT7sloadYjPmus650w5xpiYKblA>
    <xme:ervPY_-pTJMPlmQfbKMimrxA9M2sG8AzB9jnCyoB9mHrYATqiQhUo4tg8dqIXONVQ
    U31iLcSV3ntgAy5Ylk>
X-ME-Proxy-Cause: gggruggvucftvghtrhhoucdtuddrgedvhedruddvtddgvddtucetufdoteggodetrfdotf
    fvucfrrhhofhhilhgvmecuhfgrshhtofgrihhlpdfqfgfvpdfurfetoffkrfgpnffqhgen
    uceurghilhhouhhtmecufedttdenucesvcftvggtihhpihgvnhhtshculddquddttddmne
    cujfgurhepofgfggfkfffhvfevufgtsehttdertderredtnecuhfhrohhmpedfffgrvhhi
    ugcutfhhvghinhhssggvrhhgfdcuoegurghvihgusehrvggruggrhhgvrggurdgvuheqne
    cuggftrfgrthhtvghrnhepveeiffetveejgfdtteevleetleejtddvvdeftdffkeejveef
    veeguedtkefgjeevnecuvehluhhsthgvrhfuihiivgeptdenucfrrghrrghmpehmrghilh
    hfrhhomhepuggrvhhiugesrhgvrggurghhvggrugdrvghu
X-ME-Proxy: <xmx:ervPY8TT02hhkl4kBCUKqOeUqXT8jvKw5oHsWzVLmLBdrJnJgaCEOw>
    <xmx:ervPY5uuBmiSPv7WRiVrRl2IArYAgG4ntwASmneewz3WYYmvNsAstA>
    <xmx:ervPY1de3-hxqTtyw4tXQMsV7nuuGH1PEaYzm2IslwBANQDQFxZgwg>
    <xmx:e7vPY-HMpGr6ZGh_JkcEWmhnFvNZs93AeYGQ9dk1gP7-ltkOvBBGlg>
Feedback-ID: id2994666:Fastmail
Received: by mailuser.nyi.internal (Postfix, from userid 501)
        id 3334A1700089; Tue, 24 Jan 2023 06:05:30 -0500 (EST)
X-Mailer: MessagingEngine.com Webmail Interface
User-Agent: Cyrus-JMAP/3.9.0-alpha0-85-gd6d859e0cf-fm-20230116.001-gd6d859e0
Mime-Version: 1.0
Message-Id: <320c4dba-9919-404b-8a26-a8af16be1845@app.fastmail.com>
Date:   Tue, 24 Jan 2023 12:04:59 +0100
From:   "David Rheinsberg" <david@readahead.eu>
To:     linux-kernel@vger.kernel.org
Cc:     rust-for-linux@vger.kernel.org, "H. Peter Anvin" <hpa@zytor.com>,
        x86@kernel.org, "Dave Hansen" <dave.hansen@linux.intel.com>,
        "Borislav Petkov" <bp@alien8.de>, "Ingo Molnar" <mingo@redhat.com>,
        "Thomas Gleixner" <tglx@linutronix.de>
Subject: [PATCH] x86/insn_decoder_test: allow longer symbol-names
Content-Type: text/plain
Precedence: bulk
List-ID: <linux-kernel.vger.kernel.org>
X-Mailing-List: linux-kernel@vger.kernel.org

Increase the allowed line-length of the insn-decoder-test to 4k to allow
for symbol-names longer than 256 characters.

The insn-decoder-test takes objdump output as input, which may contain
symbol-names as instruction arguments. With rust-code entering the
kernel, those symbol-names will include mangled-symbols which might
exceed the current line-length-limit of the tool.

By bumping the line-length-limit of the tool to 4k, we get a reasonable
buffer for all objdump outputs I have seen so far. Unfortunately, ELF
symbol-names are not restricted in length, so technically this might
still end up failing if we encounter longer names in the future.

My compile-failure looks like this:

    arch/x86/tools/insn_decoder_test: error: malformed line 1152000:
    tBb_+0xf2>

..which overflowed by 10 characters reading this line:

    ffffffff81458193:   74 3d                   je     ffffffff814581d2 <_RNvXse_NtNtNtCshGpAVYOtgW1_4core4iter8adapters7flattenINtB5_13FlattenCompatINtNtB7_3map3MapNtNtNtBb_3str4iter5CharsNtB1v_17CharEscapeDefaultENtNtBb_4char13EscapeDefaultENtNtBb_3fmt5Debug3fmtBb_+0xf2>

Signed-off-by: David Rheinsberg <david@readahead.eu>
---
 arch/x86/tools/insn_decoder_test.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/arch/x86/tools/insn_decoder_test.c b/arch/x86/tools/insn_decoder_test.c
index 472540aeabc2..366e07546344 100644
--- a/arch/x86/tools/insn_decoder_test.c
+++ b/arch/x86/tools/insn_decoder_test.c
@@ -106,7 +106,7 @@ static void parse_args(int argc, char **argv)
 	}
 }

-#define BUFSIZE 256
+#define BUFSIZE 4096

 int main(int argc, char **argv)
 {
--
2.39.1
